numpy/ma/core.py

```diff
@@ -7505,8 +7513,7 @@ def allclose(a, b, masked_equal=True, rtol=1e-5, atol=1e-8):
         return False
     # No infs at all
     if not np.any(xinf):
-        d = filled(umath.less_equal(umath.absolute(x - y),
-                                    atol + rtol * umath.absolute(y)),
+        d = filled(less_equal(absolute(x - y), atol + rtol * absolute(y)),
```

Comment:
That's right. This way it goes through the masked ufuncs, which suppress warnings for nan at masked positions.